editors/sc-ide/core/settings/serialization.cpp

```diff
@@ -63,18 +76,25 @@ static QVariant parseTextFormat( const YAML::Node & node )
     std::string val;
     QTextCharFormat fm;
 
-    const Node ncolor = node[ "color" ];
-    if(ncolor && ncolor.IsScalar()) {
-        val = ncolor.as<std::string>();
-        fm.setForeground(QColor(val.c_str()));
-    }
-
-    const Node nbg = node[ "background" ];
-    if(nbg && nbg.IsScalar()) {
-        val = nbg.as<std::string>();
-        QColor color(val.c_str());
-        if(color.isValid())
-            fm.setBackground(color);
+    // Some really old themes (from users who've stuck around since pre-3.7 and
+    // never touched their theme) have missing color: or background: entries,
+    // which caused problems here (bug #4218). We default to black if text
+    // color is not specified.
+    QColor color = yamlNodeToColor(node["color"]);
+    fm.setForeground(color.isValid()
+        ? color
+        : Qt::black);
```

Comment:
no need to split this across multiple lines (and it's likely to be reformatted in the future anyway)